src/main/java/com/example/demo/Models/Movie.java

```diff
@@ -2,33 +2,74 @@
 
 import lombok.Data;
 
+import javax.persistence.CascadeType;
 import javax.persistence.Entity;
 import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
 import javax.persistence.Id;
-import java.util.List;
+import javax.persistence.OneToOne;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import javax.persistence.OneToMany;
+import javax.persistence.JoinColumn;
+import java.util.Set;
 
 @Data
 @Entity
 public class Movie {
-    private @Id @GeneratedValue Long id;
+    private @Id @GeneratedValue(strategy = GenerationType.IDENTITY) Long id;
+
+    @NotBlank(message = "Title is mandatory")
+    @Size(max = 50, message = "Title max length is 50 characters")
     private String title;
+
+    @NotBlank(message = "Year is mandatory")
     private String year;
+
+    @NotBlank(message = "Description is mandatory")
+    @Size(min = 100, message = "Description min length is 100 characters")
     private String description;
-    private String rate;// enum
+
+    private MovieRate rate = MovieRate.G;// enum
     private Boolean deleted;
-    /*
-     * private User registeringUser; private User updatingUser; private List<Actor>
-     * actorsList;
-     */
+
+    @JsonIgnore
+    @OneToOne(cascade = CascadeType.ALL)
```

Comment:
You are using the CascateType.ALL here what does this mean? there are other types of CascadeType could you mention the difference between them please?